src/Adapter/Shipment/QueryHandler/GetOrderShipmentsForViewingHandler.php

```diff
@@ -0,0 +1,54 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+namespace PrestaShop\PrestaShop\Adapter\Shipment\QueryHandler;
+
+use PrestaShop\PrestaShop\Core\CommandBus\Attributes\AsQueryHandler;
+use PrestaShop\PrestaShop\Core\Domain\Shipment\Exception\ShipmentNotFoundException;
+use PrestaShop\PrestaShop\Core\Domain\Shipment\Query\GetOrderShipments;
+use PrestaShop\PrestaShop\Core\Domain\Shipment\QueryHandler\GetOrderShipmentForViewingHandlerInterface;
+use PrestaShopBundle\Entity\Repository\ShipmentRepository;
+use Throwable;
+
+#[AsQueryHandler]
+class GetOrderShipmentsForViewingHandler implements GetOrderShipmentForViewingHandlerInterface
+{
+    public function __construct(
+        private readonly ShipmentRepository $repository,
+    ) {
+    }
+
+    public function handle(GetOrderShipments $query): array
```

Comment:
Query handlers must return some DTOs that are in a sub namespace `QueryResult` In this case you return a list so it could be an array od DTOs or you can create a dedicated Collection DTO for this query, which has the advantage of being clearly typed

But in both cases you can't return the Doctrine entity directly, the purpose of CQRS and domain implementation is to prevent being coupled to the persistence layer directly So you can fetch the doctrine entities, no problem with that But then they must be transformed into DTOs

And if you choose to use an array instead of a typed collection the PHPDoc must indicate the type used in the array at least